server/src/test/java/org/elasticsearch/search/SearchServiceSingleNodeTests.java

```diff
@@ -2928,6 +2928,35 @@ public void testSlicingBehaviourForParallelCollection() throws Exception {
         }
     }
 
+    public void testMaybeWrapListenerForStackTrace() {
+        // Tests that the same listener has stack trace if is not wrapped or does not have stack trace if it is wrapped.
+        SearchService service = getInstanceFromNode(SearchService.class);
+        service.getThreadPool().getThreadContext().putHeader("error_trace", "false");
+        AtomicBoolean isWrapped = new AtomicBoolean(false);
+        ActionListener<SearchPhaseResult> listener = new ActionListener<>() {
+            @Override
+            public void onResponse(SearchPhaseResult searchPhaseResult) {
+                // noop - we only care about failure scenarios
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                if (isWrapped.get()) {
+                    assertThat(e.getStackTrace().length, is(0));
+                } else {
+                    assertThat(e.getStackTrace().length, is(not(0)));
+                }
+            }
+        };
+        Exception e = new Exception();
+        e.fillInStackTrace();
+        assertThat(e.getStackTrace().length, is(not(0)));
+        listener.onFailure(e);
+        listener = service.maybeWrapListenerForStackTrace(listener);
+        isWrapped.set(true);
+        listener.onFailure(e);
```

Comment:
the functionality tested here seems easy enough to be tested without starting up a node? Could we move this test to SearchServiceTests? Does it really need a full blown instance of SearchService?

Reply:
Moved to SearchServiceTests :D